File: 03-array/02-separate-even-and-odd/02-separate-even-and-odd.rb
def separate_even_and_odd(numbers)
  odd, even = [], []
  numbers.each do |number|
    number % 2 == 0 ? even.push(number) : odd.push(number)
  end
  [odd, even]
end

print separate_even_and_odd [1,2,3,4,5]

File: 03-array/02-separate-even-and-odd/02-separate-even-and-odd.ts
const separate_even_and_odd = (numbers: number[]) => {
  return numbers.reduce<[number[], number[]]>(
    (acc, curr) => {
      const [even, odd] = acc
      if (curr % 2 === 0) even.push(curr)
      else odd.push(curr)
      return [even, odd]
    },
    [[], []]
  )
}

console.log(separate_even_and_odd([1, 2, 3, 4, 5]))
